### PR TITLE
Doc update tracing

### LINKED ISSUE #3708: README for Tracing Library
The READMEs for tracing need to be filled out.

### PR DESCRIPTION
**Pull request by samvaity**
- Package cleanup, update docs and add unit tests for tracing package.
- Update ci.yml for correct package name
- Move ProcessKind enum to util tracing 

Closes #3708

### CODE REVIEW

File: sdk/core/azure-core/src/samples/java/com/azure/core/util/tracing/TracerJavaDocCodeSnippets.java [hunk 1]
```diff
@@ -26,7 +26,9 @@ public void startTracingSpan() {
         // pass the current tracing span context to the calling method
         Context traceContext = new Context(OPENCENSUS_SPAN_KEY, "<user-current-span>");
         // start a new tracing span with the given method name and explicit parent span
-        tracer.start("azure.keyvault.secrets/setsecret", traceContext);
+        Context updatedContext = tracer.start("azure.keyvault.secrets/setsecret", traceContext);
+        System.out.printf("Span returned in the context object: %s%n",
+            updatedContext.getData(OPENCENSUS_SPAN_KEY).get().getClass());
```
Comment (conniey) [inline sdk/core/azure-core/src/samples/java/com/azure/core/util/tracing/TracerJavaDocCodeSnippets.java]: Why `.getClass()` versus just printing the object?

File: sdk/core/azure-core/src/samples/java/com/azure/core/util/tracing/TracerJavaDocCodeSnippets.java [hunk 2]
```diff
@@ -36,22 +38,27 @@ public void startTracingSpan() {
...
         // END: com.azure.core.util.tracing.start#string-context-processKind-SEND
 
         // BEGIN: com.azure.core.util.tracing.start#string-context-processKind-RECEIVE
         // start a new tracing span with explicit parent, sets the diagnostic Id (traceparent headers) on the current
         // context when process kind RECEIVE
-        Context updatedContext = tracer.start("azure.eventhubs.receive", traceContext, ProcessKind.RECEIVE);
-        System.out.printf("Diagnostic Id: {} %s%n", (String) updatedContext.getData(DIAGNOSTIC_ID_KEY).get());
+        Context updatedReceiveContext = tracer.start("azure.eventhubs.receive", traceContext,
+            ProcessKind.RECEIVE);
+        System.out.printf("Diagnostic Id: {} %s%n", (String) updatedReceiveContext.getData(DIAGNOSTIC_ID_KEY).get());
```
Comment (conniey) [inline sdk/core/azure-core/src/samples/java/com/azure/core/util/tracing/TracerJavaDocCodeSnippets.java]: Do you need to cast it to (String)? It'll do a .toString() on whatever object is returned from DIAGNOSTIC_ID_KEY.

Comment (conniey) [reply sdk/core/azure-core/src/samples/java/com/azure/core/util/tracing/TracerJavaDocCodeSnippets.java]: The {} placeholders are not required. This is only for slf4j

File: sdk/core/azure-core/src/samples/java/com/azure/core/util/tracing/TracerJavaDocCodeSnippets.java [hunk 3]
```diff
@@ -36,22 +38,27 @@ public void startTracingSpan() {
...
 
         // BEGIN: com.azure.core.util.tracing.start#string-context-processKind-PROCESS
         // start a new tracing span with remote parent and uses the span in the current context to return a scope
         // when process kind PROCESS
         Context processContext = new Context(OPENCENSUS_SPAN_KEY, "<user-current-span>")
             .addData(SPAN_CONTEXT, "<user-current-span-context>");
-        tracer.start("azure.eventhubs.process", processContext, ProcessKind.PROCESS);
+        Context updatedProcessContext = tracer.start("azure.eventhubs.process", processContext,
+            ProcessKind.PROCESS);
+        System.out.printf("Scope: {} %s%n", updatedProcessContext.getData("scope").get().getClass());
```
Comment (conniey) [inline sdk/core/azure-core/src/samples/java/com/azure/core/util/tracing/TracerJavaDocCodeSnippets.java]: {} is not required. Do you need .getClass?

Comment (samvaity) [reply sdk/core/azure-core/src/samples/java/com/azure/core/util/tracing/TracerJavaDocCodeSnippets.java]: I was just doing that to use the returned object and avoid one of our checkstyle issues.
And thought printing object won't return much information, so..

File: sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java [hunk 4]
```diff
@@ -7,14 +7,15 @@
 
 /**
  * Contract that all tracers must implement to be plug-able into the SDK.
+ *
+ * @see Tracer
```
Comment (conniey) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java]: This is a `@see` to itself?

File: sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java [hunk 5]
```diff
@@ -3,19 +3,18 @@
...
+ *
  */
 public interface Tracer {
     /**
      * Key for {@link Context} which indicates that the context contains OpenCensus span data. This span will be used
      * as the parent span for all spans the SDK creates.
      *
-     * If no span data is listed when the SDK creates its first span it will be used as the parent for all further spans
-     * it creates.
+     * If no span data is listed when the SDK creates its first span, this span key will be used as the parent span.
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java]: Wrap this in a paragraph tag.

File: sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java [hunk 6]
```diff
@@ -40,61 +38,102 @@
...
      */
     String DIAGNOSTIC_ID_KEY = "diagnostic-id";
 
     /**
      * Creates a new tracing span.
      *
-     * The {@code context} will be checked for containing information about a parent span. If a parent span is found the
-     * new span will be added as a child, otherwise the span will be created and added to the context and any downstream
-     * start calls will use the created span as the parent.
+     * The {@code context} will be checked for containing information about a parent span. If a parent span is found,
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java]: Wrap in a paragraph tag

File: sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java [hunk 7]
```diff
@@ -40,61 +38,102 @@
...
+     * @return The updated {@link Context} object containing the returned span.
+     * @throws NullPointerException if {@code methodName} or {@code context} is {@code null}.
      */
     Context start(String methodName, Context context);
 
     /**
-     * Creates a new tracing span.
+     * Creates a new tracing span for AMQP calls.
      *
      * The {@code context} will be checked for containing information about a parent span. If a parent span is found the
```
Comment (alzimmermsft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Tracer.java]: Also wrap in a paragraph tag, remember Javadocs are HTML so newlines are ignored unless it's a pre tag.

File: sdk/tracing/azure-core-tracing-opencensus/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,19 @@
-## Version 1.0.0-preview.1:
+# Release History
 
+## Version 1.0.0-preview.3 ()
```
Comment (alzimmermsft) [inline sdk/tracing/azure-core-tracing-opencensus/CHANGELOG.md]: Let's add a date here

File: sdk/tracing/azure-core-tracing-opencensus/pom.xml [hunk 9]
```diff
@@ -72,5 +77,14 @@
       <version>5.0.0-preview.4</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
```
Comment (alzimmermsft) [inline sdk/tracing/azure-core-tracing-opencensus/pom.xml]: Shouldn't have this as a non test dependency right?

Comment (samvaity) [reply sdk/tracing/azure-core-tracing-opencensus/pom.xml]: It is a test dependency.

Comment (samvaity) [reply sdk/tracing/azure-core-tracing-opencensus/pom.xml]: Didn't need this dependency, removed it.

**Pull request merged**